e2e/tests/forms/taiteen_perusop_avust.spec.ts

```diff
@@ -1,33 +1,149 @@
 
-const APPLICATION_TITLE = "Taiteen perusopetuksen avustukset"
+type UserInputData = Record<string, string>
 
-test(APPLICATION_TITLE, async ({ page }) => {
-  await selectRole(page, 'REGISTERED_COMMUNITY');
-  await startNewApplication(page, APPLICATION_TITLE)
-  await acceptCookies(page)
-  
-  // Fill step 1
-  await page.getByRole('textbox', { name: 'Hakemusta koskeva sähköposti' }).fill('asadsdqwetest@example.org');
-  await page.getByLabel('Yhteyshenkilö').fill('asddsa');
-  await page.getByLabel('Puhelinnumero').fill('0234432243');
-  await page.locator('#edit-community-address-community-address-select').selectOption({ index: 1 });
+const formInputData = {
+  additionalInformation: faker.lorem.words(),
+  attachmentInfo: faker.lorem.words(),
+  email: faker.internet.email(),
+  fullName: faker.person.fullName(),
+  phoneNumber: faker.phone.number(),
+  shortDescription: faker.lorem.words(),
+}
+
+const messageContent = faker.lorem.words();
+
+test.describe('Taiteen perusopetuksen avustukset', () => {
+  test.beforeEach(async ({ page }) => {
+    await selectRole(page, 'REGISTERED_COMMUNITY');
+    await page.goto('fi/uusi-hakemus/taide_ja_kulttuuriavustukset_tai')
+    await acceptCookies(page)
+  });
+
+  test('Submit application and send message', async ({ page }) => {
+    await fillStepOne(page);
+    await fillStepTwo(page)
+    await fillStepThree(page)
+    await fillStepFour(page)
+    await fillStepFive(page)
+    await fillStepSix(page)
+    await fillStepSeven(page)
+
+    await checkConfirmationPage(page, formInputData);
+    await submitApplication(page);
+    await checkSentApplication(page, formInputData);
+    await sendMessageToApplication(page, messageContent);
+  });
+
+  test('Application can be saved as a draft', async ({ page }) => {
+    await fillStepOne(page);
+    await saveAsDraft(page);
+
+    // Check application draft page
+    await expect(page.getByText('Luonnos')).toBeVisible()
+    await expect(page.getByRole('link', { name: 'Muokkaa hakemusta' })).toBeEnabled();
+    const applicationId = await page.locator(".webform-submission__application_id--body").innerText()
+    const pageText = await page.getByLabel('1. Hakijan tiedot').innerText();
+
+    const textsToCheck = [formInputData.email, formInputData.fullName, formInputData.phoneNumber];
+    textsToCheck.forEach(t => expect(pageText).toContain(t))
+
+    // Check if application is shown in "Keskeneräiset hakemukset"
+    await page.goto("fi/oma-asiointi")
+    const drafts = await page.locator("#oma-asiointi__drafts").innerText()
+    expect(drafts).toContain(applicationId)
+  });
+
+  test('Draft can be removed', async ({ page }) => {
+    await fillStepOne(page);
+    await page.getByRole('button', { name: 'Tallenna keskeneräisenä' }).click();
+    await page.getByRole('link', { name: 'Muokkaa hakemusta' }).click();
+    await page.getByRole('link', { name: 'Poista luonnos' }).click();
```

Comment:
Tän klikkaaminen feilaa testissä?

Reply:
`TimeoutError: locator.click: Timeout 15000ms exceeded.
=========================== logs ===========================
waiting for getByRole('link', { name: 'Poista luonnos' })
  locator resolved to <a id="webform-button--delete-draft" href="/hakemus…>…</a>
attempting click action
  waiting for element to be visible, enabled and stable
  element is visible, enabled and stable
  scrolling into view if needed
  done scrolling
  performing click action
============================================================`

Reply:
Tämän vois testata uudestaan